app/gui2/src/util/ffi.ts

```diff
@@ -0,0 +1,406 @@
+import init, { parse_to_json } from '../../rust-ffi/pkg/rust_ffi'
+import type { NonEmptyArray, Opt } from './opt'
+
+const _wasm = await init()
+
+export function parseEnso(code: string): ast.Tree {
+  const json = parse_to_json(code)
+  return JSON.parse(json)
+}
+
+export module ast {
```

Comment:
`module` -> `namespace`. also convention is to treat namespaces kinda like classes, so `PascalCase` naming